internal/proto/controller/api/resources/workers/v1/worker.proto

```diff
@@ -90,7 +90,16 @@ message Worker {
   // itself.
   WorkerConfig worker_config = 140 [json_name = "worker_config"];
 
-  // TODO: Add worker auth related fields.
+  // node_credentials_token is input only. Supports the worker led node
+  // enrollment flow where this credentials token is produced by a worker. This
+  // token is a base58 encoded types.FetchNodeCredentialsRequest from
+  // https://github.com/hashicorp/nodeenrollment
+  google.protobuf.StringValue node_credentials_token = 150 [json_name = "node_credentials_token"]; // @gotags: `class:"public"`
```

Comment:
Do we want to use `node` instead of `worker`? Also do we want `token` here 

Also given the action is `create:worker-request` maybe this should be `registration_request`? (FWIW the current value used for outputting it to the CLI is `"Worker Auth Registration Request"`

Reply:
`registration request` just seems awkward given that it's a `create worker request`.  perhaps a quick huddle and we can figure out the naming?

Reply:
we settled on `worker_auth_token`